src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderBuyAsset.tsx

```diff
@@ -142,8 +144,16 @@ export const LimitOrderBuyAsset: React.FC<LimitOrderBuyAssetProps> = memo(
         assetSymbol={asset.symbol}
         assetIcon={asset.icon}
         placeholder={isInputtingFiatSellAmount ? '$0' : '0'}
-        cryptoAmount={bnOrZero(buyAmountCryptoPrecision).isZero() ? '' : buyAmountCryptoPrecision}
-        fiatAmount={bnOrZero(buyAmountUserCurrency).isZero() ? '' : buyAmountUserCurrency}
+        cryptoAmount={
+          bnOrZero(buyAmountCryptoPrecision).isZero() && (buyAmount === '' || buyAmount === null)
+            ? ''
+            : buyAmountCryptoPrecision
+        }
+        fiatAmount={
+          bnOrZero(buyAmountUserCurrency).isZero() && (buyAmount === '' || buyAmount === null)
+            ? ''
+            : buyAmountUserCurrency
```

Comment:
Can we extract these to xyzFormatted constants for readibility?

Also can probably cleanup checks there with an early return instead of a ternary 🙏🏽

Reply:
done in [c3f212c](https://github.com/shapeshift/web/pull/9052/commits/c3f212c6ff94d2d0a52075cf4ff77ea9c5efc3b3)